api/tasks.go

```diff
@@ -737,9 +737,6 @@ func (tmpl *Template) Canonicalize() {
 	if tmpl.Envvars == nil {
 		tmpl.Envvars = boolToPtr(false)
 	}
-	if tmpl.VaultGrace == nil {
-		tmpl.VaultGrace = timeToPtr(15 * time.Second)
```

Comment:
Removing this is probably the source of panics/issues. It ensured VaultGrace was always initialized (never nil) in the HTTP API.

Removing this code is the right thing to do. We just need to fix any code that depended on VaultGrace always being non-nil. I'll investigate and try to help.

Reply:
Sorry haven't found time today to fix this up. The really sneaky place that converts public `api` structs to internal `nomad/structs` structs is hidden in the job endpoint here: https://github.com/hashicorp/nomad/blob/v0.10.4/command/agent/job_endpoint.go#L620

That may need updating to convert `nil` api VaultGrace values to `0` nomad/struct VaultGrace values.

Reply:
See comment below.